src/platform/src/linux/mod.rs

```diff
@@ -159,6 +159,10 @@ pub fn unlink(path: *const c_char) -> c_int {
     e(unsafe { syscall!(UNLINKAT, AT_FDCWD, path, 0) }) as c_int
 }
 
+pub fn waitpid(pid: pid_t, stat_loc: *mut c_int, options: c_int) -> pid_t {
+    e(unsafe { syscall!(WAIT4, pid, stat_loc, options) }) as pid_t
```

Comment:
This should be `syscall(WAIT4, pid, stat_loc, options, 0)`. The `wait4` syscall take 4 params. The fourth being a pointer to a `struct rusage`, which is null in the default case.